[docs] Split pattern search order (#5693)
huggingface/datasets: docs/source/repository_structure.mdx

└── test.csv
```

## Split pattern hierarchy

🤗 Datasets searches for certain patterns in the dataset repository to automatically infer the dataset splits. There is an order to the patterns, beginning with the custom filename split format to treating all files as a single split if no pattern is found.

### Custom filename split

> The most flexible pattern to accomodate other split types in addition to the traditional train, validation, and test sets, is to name your data files like `data/<split_name>-xxxxx-of-xxxxx.csv`.

Reviewer: ```suggestion
If your dataset has split that have non traditional names, you should name your data files like `data/<split_name>-xxxxx-of-xxxxx.csv`.
```
Not the best wording in my suggestion, feel free to change, I just wanted to point out that I wouldn't encourage for this pattern, I mean no need to name it the most flexible. it's quite a specific one. Putting all the data into a single directory named as a split (the next pattern) is easier and for most cases enough.

    └── validation.csv
```

### Single split

> When 🤗 Datasets can't find any of the above patterns, then it'll treat all the files as a single train split. If your dataset splits aren't loading as expected, this may be why!

Reviewer: ```suggestion
When 🤗 Datasets can't find any of the above patterns, then it'll treat all the files as a single train split. If your dataset splits aren't loading as expected, it may be due to incorrect pattern.
```

## Split name keywords

> There are several acceptable ways to name splits. Validation splits are sometimes called "dev", and test splits may be referred to as "eval".

Reviewer: ```suggestion
There are several ways to name splits. Validation splits are sometimes called "dev", and test splits may be referred to as "eval".
```
